### ISSUE TITLE
AWS credentials need to be moved to a keystore

### ISSUE THREAD

**Issue by bploetz** [crate/crate#7009]
**CrateDB version**: 2.3.3

**JVM version**: JVM[Oracle Corporation/OpenJDK 64-Bit Server VM/1.8.0_151/25.151-b12]

**OS version / environment description**: Linux/4.4.51-40.58.amzn1.x86_64/amd64 (Amazon ECS optimized AMI)

**Problem description**:
Per the documentation (https://crate.io/docs/crate/reference/en/latest/config/cluster.html#discovery-on-amazon-ec2) in order to use EC2 discovery (and S3 for snapshots) you must include your AWS access key and secret key in the config via the `cloud.aws.access_key` and `cloud.aws.secret_key` properties. However, doing so leads to a warning in the logs upon startup:

```
[2018-03-06T15:20:00,401][WARN ][o.e.d.c.s.Settings       ] [cloud.aws.access_key] setting was deprecated in Elasticsearch and will be removed in a future release! See the breaking changes documentation for the next major version.
[2018-03-06T15:20:00,401][WARN ][o.e.d.c.s.Settings       ] [cloud.aws.secret_key] setting was deprecated in Elasticsearch and will be removed in a future release! See the breaking changes documentation for the next major version.
```

It looks like Elasticsearch has moved to storing sensitive config values in the Elasticsearch keystore (reference: https://discuss.elastic.co/t/s3-repository-deprecated-settings-warning/92354/2). When I change `cloud.aws.access_key` and `cloud.aws.secret_key` to their new values `discovery.ec2.access_key` and `discovery.ec2.secret_key`, I get the following exception when starting CrateDB:

```
[2018-03-06T16:23:31,787][WARN ][o.e.b.ElasticsearchUncaughtExceptionHandler] [Mangart] uncaught exception in thread [main]
org.elasticsearch.bootstrap.StartupException: java.lang.IllegalArgumentException: Setting [discovery.ec2.access_key] is a secure setting and must be stored inside the Elasticsearch keystore, but was found inside elasticsearch.yml
    at org.elasticsearch.bootstrap.StartupExceptionProxy.<init>(StartupExceptionProxy.java:31) ~[crate-app-2.3.3.jar:2.3.3]
    at io.crate.bootstrap.CrateDB.init(CrateDB.java:142) ~[crate-app-2.3.3.jar:2.3.3]
    at io.crate.bootstrap.CrateDB.execute(CrateDB.java:118) ~[crate-app-2.3.3.jar:2.3.3]
    at org.elasticsearch.cli.EnvironmentAwareCommand.execute(EnvironmentAwareCommand.java:76) ~[crate-app-2.3.3.jar:2.3.3]
    at org.elasticsearch.cli.Command.mainWithoutErrorHandling(Command.java:134) ~[crate-app-2.3.3.jar:2.3.3]
    at org.elasticsearch.cli.Command.main(Command.java:90) ~[crate-app-2.3.3.jar:2.3.3]
    at io.crate.bootstrap.CrateDB.main(CrateDB.java:87) ~[crate-app-2.3.3.jar:2.3.3]
    at io.crate.bootstrap.CrateDB.main(CrateDB.java:80) ~[crate-app-2.3.3.jar:2.3.3]
```

So these properties need to change, but it looks like CrateDB needs to be updated to use keystores for storing sensitive configs like this.

**Steps to reproduce**:
Configure `cloud.aws.access_key` and `cloud.aws.secret_key` respectively.

**Comment by mfussenegger** [comment]
Thanks for the report. We'll take a closer look at this.

**Comment by andreidan** [comment]
@bploetz We addressed these settings issues and both `discovery.ec2.access_key` and `discovery.ec2.secret_key` will be available in the next major release (Crate 3.0). 

We looked into storing the settings in a keystore like ES does but we decided to continue to allow them to be set as they are now (yml file, vm args etc) as the S3 settings are used in repositories and having them in a keystore would essentially make the repositories unusable after an upgrade to Crate 3.0.
Also Docker users would have to bake the key management into the docker images.

In the future we'll revisit this, but for now, after updating to Crate 3.0 everything related to settings will work as with current versions.
